general-contribution-guideline.md

```diff
@@ -0,0 +1,69 @@
+# Contribution Guidelines (general)
+
+This guide describes the process and the conventions for contributing an article without relaying too much on GitHub. For detailed instructions on how to contribute using GitHub, please refer to the [GitHub Contribution Guidelines](./github-contribution-guidelines.md). 
+
+## Announce
+
+1. **Check for Duplicates:** Is there already an article about your topic?
+  
+   * Please check our [Backlog of Articles](https://github.com/Zuehlke/zuehlke-book/issues?q=is%3Aissue+label%3Aarticle) whether there is already an article existing or planned for similar topic. You need a GitHub account for this.
+  
+   * If there is already a similar article, you can contact the author and find a solution (leave the subject to the author, offer your help as reviewer or co-author, take the subject over, or write a second article in response or extension to the first article, ...). 
+   
+2. **Announce an Article:**
+  
+   * Open a [New Issue for your Article](https://github.com/Zuehlke/zuehlke-book/issues/new?labels=article&title=Article:%20%3Cput%20topic%20title%20here%3E)
+  
+   * Describe roughly the idea you have in mind, what the article should be about.
+
+## Discuss
+
+It proved to be useful to discuss about an articles content before writing the article. Ask your colleagues, maybe they have some valuable input! 
+
+To discuss about an articles content, you have various options:
+
+   * Use the [Issues on Github](https://github.com/Zuehlke/zuehlke-book/issues): discuss directly in the issue you created - and don't miss to give feedback to other announced articles.
+
+   * Use Yammer, a coffee break, a team event or similar
+
+## Write
+
+**Content Conventions:**
+
+* written in English
+
+* short enough to be read in 10 minutes: size should be less than 2 A4 pages long in printout PDF version (that is between 400 and 750 words)
+
+* reflects on a single subject only (better make several articles, if you have too much material / ideas to write about)
+
+* matches our [Vision](./README.md#vision): it is about practicaly proven concepts, best practices, cultural philosophies or great ideas that we use in our daily work at Zühlke.
+
+* the subject is related to our work and the advice concrete enough to consider.
+
+* the subject is something you have experience on in some real projects and not just theoretical ideas
+
+**Language Conventions:**
+
+@Rolf: Please add title guideline
+
+**Formal Conventions:**
+
+* Textfile with the file extension *.md
+
+* Formatted using the [Markdown Syntax](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
+
+The [Example Article](https://raw.githubusercontent.com/Zuehlke/zuehlke-book/develop/articles/_example.md) is a good starting point.
```

Comment:
I propose to only uses a MASTER branch in the repo. Having a DEVELOP branch is an additional hurdle / source of error with no substantial benefit. 
@bruderol can I remove the develop branch and adjust the instructions?

Reply:
Simple rule of thumb: If you are deploying from source use a develop/master dual branch strategy for an additional buffer of safety to live production.

In any other situation you add unnecessary complexity.
We don't even have a compilation process yet, let alone a deployment/publishing target. Adding an extra mirror to master IMHO is unnecessary.

Reply:
I totally second that :)

Reply:
Disagree - develop/master separation has become a defacto standard. this will be useful to control when we publish something once we have an automated build to publish the master branch. master branch will be read only, so nobody can or will work on it, so there is no additional source of error. If cloning the project it will checkout develop by default. and also PRs will be done by default to that branch. So where do you see the problem? I prefer to have this separation in all repositories on github. If you introduce it later (and some people forget to switch) then you have an additional source of error.

I also not understand why the link above is so complicated - why dont you just use relative links (then it will always point to the branch you are currently on)?